src/Shared/Buffers.MemoryPool/MemoryPoolBlock.cs

```diff
@@ -6,52 +6,29 @@
 namespace System.Buffers
 {
     /// <summary>
-    /// Block tracking object used by the byte buffer memory pool. A slab is a large allocation which is divided into smaller blocks. The
-    /// individual blocks are then treated as independent array segments.
+    /// Wraps an array allocated in the pinned object heap in a reusable block of managed memory
     /// </summary>
     internal sealed class MemoryPoolBlock : IMemoryOwner<byte>
     {
-        private readonly int _offset;
-        private readonly int _length;
-
-        /// <summary>
-        /// This object cannot be instantiated outside of the static Create method
-        /// </summary>
-        internal MemoryPoolBlock(SlabMemoryPool pool, MemoryPoolSlab slab, int offset, int length)
+        internal MemoryPoolBlock(SlabMemoryPool pool, int length)
         {
-            _offset = offset;
-            _length = length;
-
             Pool = pool;
-            Slab = slab;
 
-            Memory = MemoryMarshal.CreateFromPinnedArray(slab.PinnedArray, _offset, _length);
+            var pinnedArray = GC.AllocateUninitializedArray<byte>(length, pinned: true);
+
+            Memory = MemoryMarshal.CreateFromPinnedArray(pinnedArray, 0, pinnedArray.Length);
         }
 
         /// <summary>
         /// Back-reference to the memory pool which this block was allocated from. It may only be returned to this pool.
         /// </summary>
         public SlabMemoryPool Pool { get; }
```

Comment:
Remove? Only spot I see this used is when `BLOCK_LEASE_TRACKING` is defined.

Reply:
Its used in dispose

Reply:
D'oh